Vagrantfile

```diff
@@ -51,16 +51,29 @@ def vm_cpus
   $vb_cpus.nil? ? $vm_cpus : $vb_cpus
 end
 
+# Make variables from contrib/utils.sh accessible
+if File.exists?(CONTRIB_UTILS_PATH)
+  cu_vars = Hash.new do |hash, key|
+    value = `. #{CONTRIB_UTILS_PATH} 2> /dev/null && echo $#{key}`.chomp
+    hash[key] = value unless value.empty?
+  end
+else
+  raise Vagrant::Errors::VagrantError.new, "The file '#{CONTRIB_UTILS_PATH}' is missing."
+end
+
+coreos_channel = cu_vars["COREOS_CHANNEL"]
```

Comment:
Could this be preserved as `$update_channel`? One of the (unstated) goals here is to minimize differences with the upstream Vagrantfile in https://github.com/coreos/coreos-vagrant, and if we keep that variable name the diff is just a bit smaller. Otherwise this looks great.